capstone-backend/src/test/java/com/google/sps/UserTest.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.google.sps.data.User;
+
+/** */
+@RunWith(JUnit4.class)
+public final class UserTest {
+  private User userOne;
+  private User userTwo;
+
+  private static final String EMAIL_ONE = "emailOne";
+  private static final String EMAIL_TWO = "emailTwo";
+  private static final String NEW_EMAIL = "newEmail";
+
+  private static final String USERNAME_ONE = "usernameOne";
+  private static final String USERNAME_TWO = "usernameTwo";
+  private static final String NEW_USERNAME = "newUsername";
+
+
+  @Before
+  public void setUp() {
+    userOne = new User(1, EMAIL_ONE, USERNAME_ONE);
+    userTwo = new User(2, EMAIL_TWO, USERNAME_TWO);
+  }
+  
+  @Test
+  public void testConstructor() {
+    Assert.assertEquals(userOne.getID(), 1);
+    Assert.assertEquals(userOne.getEmail(), EMAIL_ONE);
+    Assert.assertEquals(userOne.getUsername(), USERNAME_ONE);
+    Assert.assertEquals(userOne.getFriends().size(), 0);
+
+    Assert.assertEquals(userTwo.getID(), 2);
+    Assert.assertEquals(userTwo.getEmail(), EMAIL_TWO);
+    Assert.assertEquals(userTwo.getUsername(), USERNAME_TWO);
+    Assert.assertEquals(userTwo.getFriends().size(), 0);
+  }
+
+  @Test 
+  public void testEmail() {
+    userOne.setEmail(NEW_EMAIL);
+    Assert.assertEquals(userOne.getEmail(), NEW_EMAIL);
+    Assert.assertEquals(userTwo.getEmail(), EMAIL_TWO);
+  }
+
+  @Test 
+  public void testUsername() {
+    userOne.setUsername(NEW_USERNAME);
+    Assert.assertEquals(userOne.getUsername(), NEW_USERNAME);
+    Assert.assertEquals(userTwo.getUsername(), USERNAME_TWO);
+  }
+
+  @Test 
+  public void testAddFriend() {
+    userOne.addFriend(new Long(3));
```

Comment:
Here and below, I'd recommend using the primitive long (i.e., `userOne.addFriend(3L)`) unless you need to use the Object form (Long) (e.g.: https://stackoverflow.com/questions/5857812/long-vs-integer-long-vs-int-what-to-use-and-when).

Reply:
We originally tried that, but assertEquals threw the following error: 

```UserTest.java:[65,13] reference to assertEquals is ambiguous
[ERROR]   both method assertEquals(java.lang.Object,java.lang.Object) in org.junit.Assert and method assertEquals(long,long) in org.junit.Assert match```

So we decided to use the Object form rather than the primitive, unless there's a better workaround we should use instead.

Reply:
Got it, I think it's because of: `for (Long l : userOne.getFriends())` (so `l` is a `Long`). You'll need to either pass the arguments to Assert.assertEquals as `Long` or both arguments as `long`. I think this might clarify what's going on here (https://www.btaz.com/java/junit-4-error-reference-to-assertequals-is-ambiguous/).